test/action_test.dart

```diff
@@ -201,16 +199,231 @@ void main() {
 
         Completer syncCompleter;
         Completer asyncCompleter;
-        var action = Action();
-        action.listen((_) => syncCompleter.complete());
+        var action = Action()
+          ..listen((_) => syncCompleter.complete())
+          ..listen((_) async {
+            asyncCompleter.complete();
+          });
+        stopwatch.start();
+        for (var i = 0; i < sampleSize; i++) {
+          syncCompleter = Completer();
+          asyncCompleter = Completer();
+          await action();
+          await Future.wait([syncCompleter.future, asyncCompleter.future]);
+        }
+        stopwatch.stop();
+        var averageStreamDispatchTime =
+            stopwatch.elapsedMicroseconds / sampleSize / 1000.0;
+
+        print('awaitable action (ms): $averageActionDispatchTime; '
+            'stream-based action (ms): $averageStreamDispatchTime');
+      }, skip: true);
+    });
+  });
+
+  group('Action2', () {
+    Action2<String> action;
+
+    setUp(() {
+      action = Action2<String>();
+    });
+
+    test('should only be equivalent to itself', () {
+      Action2 _action = Action2();
+      Action2 _action2 = Action2();
+      expect(_action == _action, isTrue);
+      expect(_action == _action2, isFalse);
+    });
+
+    test('should support dispatch with a payload', () async {
+      Completer c = Completer();
+      action.listen((payload) {
+        expect(payload, equals('990 guerrero'));
+        c.complete();
+      });
+
+      await action('990 guerrero');
+      return c.future;
+    });
+
+    test('should dispatch by default when called', () async {
+      Completer c = Completer();
+      action.listen((payload) {
+        expect(payload, equals('990 guerrero'));
+        c.complete();
+      });
+
+      await action('990 guerrero');
+      return c.future;
+    });
```

Comment:
Hmm.. I know this is copied from the test suite above, but in both cases this test seems identical to the one before it.